### ISSUE TITLE
[Question] BTRFS-mode, /Home contains sytem & app config files by default

### ISSUE THREAD

**Issue by zilexa** [teejee2008/timeshift#692]
Ubuntu Budgie 20.10, BTRFS with @/ and @/Home subvolumes (defaults). 

I previously used EXT4 and RSync-mode, excluding personal folders and temp folders:
```
    "/home/asterix/.config/syncthing/index-v0.14.0.db/***",
    "/home/asterix/.cache/***",
    "/home/asterix/Documents/***",
    "/home/asterix/Downloads/***",
    "/home/asterix/Media/***",
    "/home/asterix/Music/***",
    "/home/asterix/Photos/***",
    "+ /home/asterix/**",
    "/root/**"
```
I restored rsync snapshots frequently and it worked perfectly: a fresh system while still having the latest versions/all personal files.

The rest of the /home folder actually contains lots of configuration files, for both the system GUI of the OS and for basically all applications. They cannot be considered personal files. Excluding the entire /home folder would not make sense if you need a complete system backup experience.

With BTRFS, I cannot exclude these folders. This means, if I restore a BTRFS snapshot, I loose any new personal documents and get old version of existing personal documents. 

I cannot imagine this is by design. Or is this really based on the assumption /home only contains personal files? Isn't that incorrect?
 I don't believe there is an Ubuntu distribution that does that. /home actually contains lots of important system and application files, it can even contain entire applications. 

I love BTRFS, but this is quite a disturbing issue, loosing your actual personal files when restoring a snapshot.

**Comment by wmutschl** [comment]
You could create another btrfs subvolume, let's call it `mydata` and move your personal folders and temp folders there. Then, if you restore both @ and @home, you should get what you aim for, right?

In more detail, let's assume your btrfs root partition is `sda3`, then I would do something like this:
```sh
mount -o subvolid=5 /dev/sda3 /mnt
ls /mnt
# @ @home
btrfs subvolume create /mnt/mydata
mv /mnt/@home/$USER/Documents /mnt/mydata/ # move any other data you would like to exclude to mydata
# symlink back into home
ln -s /mnt/mydata/Documents /mnt/@home/$USER/Documents
```
Alternatively, you could adjust the mount points in your fstab...

Alternatively, note that if you do snapshots of both @ and @home you are always on the save side. For example, if you then decide to restore to a certain point in time, timeshift does not delete the "bad" @ and @home subvolumes. So what I always do is to snapshot both @ and @home automatically, but restore @ only with timeshift. When I need some files from the past in @home, I simply mount my topvolume and copy these over.

**Comment by zilexa** [comment]
Thanks! I have been reading some more and indeed, creating subvolume(s) is the way to go. Now I guess there are 2 approaches. 

1. Your approach, create a single new subvolume @mydata (/mnt/data) with the personal folders to it, symlink those folders to the right location. 
2. Create a subvolume for each personal folder + .cache folder, 6 in total. This eliminates the need for symlinks.

In both cases, snapshot @ and @home, but not the newly created subvolume(s). 
I like (2) in this case as I only use symlinks when needed. And it seems when using BTRFS it is meant to exclude stuff via subvolumes.

**Comment by wmutschl** [comment]
No, it won't include it as you have a flat btrfs subvolume structure, all is good then. Also nested subvolumes will be ignored. Anyways, you should differentiate between the btrfs root filesystem (which has always subvolid 5) and the root mount point of your system, which is mounted to a subvolume called @.
This thread by @TomZ over at the old Manjaro forum helped me understand this, maybe it is of use to you as well:
https://archived.forum.manjaro.org/t/btrfs-tips-and-tricks/71186/6

**Comment by zilexa** [comment]
Is there a reason you symlink?
> ```
> # symlink back into home
> ln -s /mnt/mydata/Documents /mnt/@home/$USER/Documents
> ```

Instead of just mounting those folders back into home? I read in an 8-year old comment when you move files between subvolumes, they will get physically copied because BTRFS treats subvolumes as seperate partitions. That would not be desirable and I would definitely go for symlinks. 

Unless it is possible to mount a subfolder of a subvolume? I could not find this in any documentation, I doubt it is possible. Say your subvolume is @mydata, now I want to mount a folder inside @mydata:
`sudo mount -o subvol=/@mydata/Documents /dev/sda3 /home/Documents`

**Comment by wmutschl** [comment]
The cleanest way is to mount the different subvolumes via your fstab.
